src/monmove.cpp

```diff
@@ -387,7 +387,8 @@ void monster::plan()
         }
     } else if( friendly != 0 && !docile ) {
         for( monster &tmp : g->all_monsters() ) {
-            if( tmp.friendly == 0 && seen_levels.test( tmp.pos().z + OVERMAP_DEPTH ) ) {
+            if( tmp.friendly == 0 && tmp.attitude_to( *this ) == Attitude::HOSTILE &&
+                seen_levels.test( tmp.pos().z + OVERMAP_DEPTH ) ) {
```

Comment:
```suggestion
                 !has_flag( MF_PET_WONT_FOLLOW )  && seen_levels.test( tmp.pos().z + OVERMAP_DEPTH ) ) {
```
And also remove the incorrect use of PET_WONT_FOLLOW in line 345 (in the section for not-friendly monsters).